mlflow/models/evaluation/base.py

```diff
@@ -33,6 +33,104 @@
 _logger = logging.getLogger(__name__)
 
 
+class EvaluationMetric:
+    """
+    A model evaluation metric.
+
+    :param eval_fn:
+        A function that computes the metric with the following signature:
+
+        .. code-block:: python
+
+            def eval_fn(
+                eval_df: Union[pandas.Dataframe, pyspark.sql.DataFrame],
+                builtin_metrics: Dict[str, float],
+            ) -> float:
+                \"\"\"
+                :param eval_df:
+                    A Pandas or Spark DataFrame containing ``prediction`` and ``target`` column.
+                    The ``prediction`` column contains the predictions made by the model.
+                    The ``target`` column contains the corresponding labels to the predictions made
+                    on that row.
+                :param builtin_metrics:
+                    A dictionary containing the metrics calculated by the default evaluator.
+                    The keys are the names of the metrics and the values are the scalar values of
+                    the metrics. Refer to the DefaultEvaluator behavior section for what metrics
+                    will be returned based on the type of model (i.e. classifier or regressor).
+                :return:
+                    The metric value.
+                \"\"\"
+                ...
+
+    :param name: The name of the metric.
+    :param greater_is_better: Whether a higher value of the metric is better.
+    :param long_name: (Optional) The long name of the metric. For example,
+        ``"root_mean_squared_error"`` for ``"mse"``.
+    """
+
+    def __init__(self, eval_fn, name, greater_is_better, long_name=None):
+        self.eval_fn = eval_fn
+        self.name = name
+        self.greater_is_better = greater_is_better
+        self.long_name = long_name or name
+
+    def __str__(self):
+        if self.long_name:
+            return (
+                f"EvaluationMetric(name={self.name}, long_name={self.long_name}, "
+                f"greater_is_better={self.greater_is_better})"
+            )
+        else:
+            return f"EvaluationMetric(name={self.name}, greater_is_better={self.greater_is_better})"
+
+
+def make_metric(
+    eval_fn,
+    name,
+    greater_is_better,
+    long_name=None,
+):
```

Comment:
@dbczumar 

```suggestion
def make_metric(
    eval_fn,
    greater_is_better,
    name=None,
    long_name=None,
):
```

I think it'd be helpful if `make_metric`infers the metric name from the given function.

```python
def my_metric():
    ...

make_metric(my_metric, greater_is_better=True)
# -> "my_metric" (inferred from the `__name__` attribute)
make_metric(lambda df, metrics: 0, greater_is_better=True)  # -> can't infert the name, throw.
```

What do you think?

Reply:
Brilliant! Yes :)